llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4004,6 +4004,19 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
 
     break;
   }
+  case ISD::UINT_TO_FP: {
```

Comment:
Do we do the same in the IR already? Would be good to do the same for globalisel

Reply:
We currently don't track known bits for FP types on the IR level at all: https://github.com/llvm/llvm-project/blob/d6ded91121fa02837ef6c8c7f06d98ccf4a0fe16/llvm/lib/Analysis/ValueTracking.cpp#L1761-L1762

Reply:
Is there an inherent reason for that?

Reply:
We used to pretend float didn't mean IEEE float layout, but I think that's behind us now